crates/bevy_core/src/lib.rs

```diff
@@ -57,7 +57,9 @@ fn register_rust_types(app: &mut App) {
         .register_type::<HashSet<String>>()
         .register_type::<Option<String>>()
         .register_type::<Option<bool>>()
+        .register_type::<Option<f32>>()
```

Comment:
Non-blocking: While we're here, we could also consider registering the `ReflectSerialize` and `ReflectDeserialize` for all of these `Option` types.

Reply:
Also, it seems like the `GetTypeRegistration` impl for `Option<T>` doesn't register `ReflectDefault` when it definitely could. If you want, you could add that as well. But definitely not necessary for the sake of this PR.